src/main/java/uk/gov/ons/ctp/response/action/message/CaseNotificationReceiver.java

```diff
@@ -1,58 +1,67 @@
-
-  @Qualifier("genericRabbitTemplate")
-  @Autowired
-  private RabbitTemplate rabbitTemplate;
-
+  @Autowired private PubSub pubSub;
   @Autowired private ObjectMapper objectMapper;
+  @Autowired private CaseNotificationService caseNotificationService;
 
-  private static final String ACTION_OUTBOUND_EXCHANGE = "action-outbound-exchange";
-  private static final String ACTION_ROUTING_KEY = "Action.CaseNotificationHandled.binding";
-
-  @ServiceActivator(
-      inputChannel = "caseNotificationTransformed",
-      adviceChain = "caseNotificationRetryAdvice")
-  public void acceptNotification(final CaseNotification caseNotification) throws CTPException {
+  private static ProjectSubscriptionName getPubSubSubscription(
+      String project, String subscriptionId) {
+    log.with(subscriptionId, project)
+        .info("creating pubsub subscription for case notification with subscriptionId in project");
+    ProjectSubscriptionName subscriptionName = ProjectSubscriptionName.of(project, subscriptionId);
+    return subscriptionName;
+  }
 
-    try {
-      log.with("case_id", caseNotification.getCaseId())
-          .info("Receiving case notification for case id");
-      caseNotificationService.acceptNotification(caseNotification);
-    } finally {
-      try {
-        rabbitTemplate.convertAndSend(
-            ACTION_OUTBOUND_EXCHANGE,
-            ACTION_ROUTING_KEY,
-            objectMapper.writeValueAsString(caseNotification));
-        log.with("exchange", ACTION_OUTBOUND_EXCHANGE)
-            .with("case_id", caseNotification.getCaseId())
-            .with("case_notification", caseNotification)
-            .debug("Just wrote rabbit message to rabbit");
-      } catch (JsonProcessingException e) {
-        log.error("Can't send message", e);
-      }
-    }
+  /**
+   * This method provides an Event Listener subscription for the action case notification.
+   *
+   * @returns void
+   * @throws IOException
+   */
+  @EventListener(ApplicationReadyEvent.class)
+  public void caseNotificationSubscription() throws IOException {
+    // Instantiate an asynchronous message receiver.
+    MessageReceiver receiver =
+        (PubsubMessage message, AckReplyConsumer consumer) -> {
+          // Handle incoming message, then ack the received message.
+          log.with(message.getMessageId()).info("Receiving message ID from PubSub");
+          log.with(message.getData().toString()).debug("Receiving data from PubSub ");
+          try {
+            CaseNotification caseNotificationReceiver =
+                objectMapper.readValue(message.getData().toStringUtf8(), CaseNotification.class);
+            caseNotificationService.acceptNotification(caseNotificationReceiver);
+            consumer.ack();
+          } catch (final IOException | CTPException e) {
+            log.with(e)
+                .error(
+                    "Something went wrong while processing message received from PubSub for case notification");
```

Comment:
should it nack() the message here?

Reply:
negative ack.? don't think so I am expecting IOException or CTP exception to happen by line number 53 and the ack is line 54

Reply:
Ya I need to do that